spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -92,6 +97,10 @@
 
 	private final Map<String, RedisLock> locks = new ConcurrentHashMap<>();
 
+	private final ConcurrentLinkedDeque<String> queue = new ConcurrentLinkedDeque<>();
```

Comment:
I don't think we need to worry about `put` and `remove` optimization.
Those `obtain()` and `expireUnusedOlderThan()` are not designed to be used in the heavy concurrent environment.
The first one is just for lock reference, which you can store for reuse.
The second one indeed has to be called on from an exclusive thread.

Probably for a solid logic it is enough to make those to method as `synchronized`, if you don't want to wrap `LinkedHashMap` into a `Collections.synchronizedMap()`.

And we might need to unit-test to cover this new functionality.

Reply:
Thank you for your review.
Fixed to use `LinkedHashMap`.
and
Added test case.